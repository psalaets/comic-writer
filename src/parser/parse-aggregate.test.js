import {
  parseSpreadContent,
  parsePreSpreadLines
} from './parse';

describe('parse aggregate', () => {
  describe('parseRawSpreadChunk', () => {
    test('basic', () => {
      const spreadLine = 'Page 1';
      const childLines = [
        'Panel 1',
        'Establihsing shot of Manhattan. There is a helicopter hovering above a skyscraper.'
      ];

      const result = parseSpreadContent({
        spread: spreadLine,
        children: childLines
      });

      expect(result).toMatchSnapshot();
    });

    test('multiple panels', () => {
      const spreadLine = 'Page 1';
      const childLines = [
        'Panel 1',
        'Establihsing shot of Manhattan. There is a helicopter hovering above a skyscraper.',
        '',
        'Panel 2',
        'The character walks down the street. Hands stuffed in pockets.',
        '',
        'Panel 3',
        'Character looks behind him, thinking he is being followed.',
        ''
      ];

      const result = parseSpreadContent({
        spread: spreadLine,
        children: childLines
      });

      expect(result).toMatchSnapshot();
    });

    test('multiple letterings', () => {
      const spreadLine = 'Page 1';
      const childLines = [
        'Panel 1',
        'Establihsing shot of NYC. There is a helicopter hovering above a skyscraper.',
        '',
        'Panel 2',
        'The character walks down the street. Hands stuffed in pockets.',
        '\tCAPTION: They\'re coming after me. I can feel it.',
        '\tCAPTION: Am I paranoid?',
        '',
        'Panel 3',
        'Character looks behind him, thinking he is being followed.',
        '\tCAPTION: I like to call it prepared.',
        '',
        'Panel 4',
        'Character running frantically.',
        '\tCHARACTER: Help! Help!',
        '\tCAPTION: Nobody cares, not here.',
        'Panel 5',
        'Character is shot in the back.',
        '\tSFX: BLAM!',
        '\tCAPTION: Bout damn time.'
      ];

      const result = parseSpreadContent({
        spread: spreadLine,
        children: childLines
      });

      expect(result).toMatchSnapshot();
    });
  });

  describe('parsePreSpreadLines', () => {
    test('metadata, blank lines and paragraphs', () => {
      const lines = [
        'Title: The Comic',
        '',
        'This is a comic about some people. It is my first comic.',
        '',
        'Issue: 5',
        ''
      ];

      const result = parsePreSpreadLines(lines);

      expect(result).toMatchSnapshot();
    });

    test('empty', () => {
      const lines = [];

      const result = parsePreSpreadLines(lines);

      expect(result).toMatchSnapshot();
    });
  });
});