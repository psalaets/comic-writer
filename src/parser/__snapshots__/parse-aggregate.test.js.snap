// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`parse aggregate parsePreSpreadLines empty 1`] = `Array []`;

exports[`parse aggregate parsePreSpreadLines metadata, blank lines and paragraphs 1`] = `
Array [
  Object {
    "name": "Title",
    "type": "metadata",
    "value": "The Comic",
  },
  Object {
    "type": "blank",
  },
  Object {
    "content": "This is a comic about some people. It is my first comic.",
    "type": "paragraph",
  },
  Object {
    "type": "blank",
  },
  Object {
    "name": "Issue",
    "type": "metadata",
    "value": "5",
  },
  Object {
    "type": "blank",
  },
]
`;

exports[`parse aggregate parseRawSpreadChunk basic 1`] = `
Object {
  "children": Array [
    Object {
      "captionCount": 0,
      "captionWordCount": 0,
      "dialogueCount": 0,
      "dialogueWordCount": 0,
      "number": 1,
      "sfxCount": 0,
      "speakers": Array [],
      "type": "panel",
    },
    Object {
      "content": "Establihsing shot of Manhattan. There is a helicopter hovering above a skyscraper.",
      "type": "paragraph",
    },
  ],
  "spread": Object {
    "captionCount": 0,
    "captionWordCount": 0,
    "dialogueCount": 0,
    "dialogueWordCount": 0,
    "pageCount": 1,
    "panelCount": 1,
    "sfxCount": 0,
    "speakers": Array [],
    "type": "spread",
  },
}
`;

exports[`parse aggregate parseRawSpreadChunk multiple letterings 1`] = `
Object {
  "children": Array [
    Object {
      "captionCount": 0,
      "captionWordCount": 0,
      "dialogueCount": 0,
      "dialogueWordCount": 0,
      "number": 1,
      "sfxCount": 0,
      "speakers": Array [],
      "type": "panel",
    },
    Object {
      "content": "Establihsing shot of NYC. There is a helicopter hovering above a skyscraper.",
      "type": "paragraph",
    },
    Object {
      "type": "blank",
    },
    Object {
      "captionCount": 2,
      "captionWordCount": 11,
      "dialogueCount": 0,
      "dialogueWordCount": 0,
      "number": 2,
      "sfxCount": 0,
      "speakers": Array [],
      "type": "panel",
    },
    Object {
      "content": "The character walks down the street. Hands stuffed in pockets.",
      "type": "paragraph",
    },
    Object {
      "content": Array [
        Object {
          "content": "They're coming after me. I can feel it.",
          "type": "text",
        },
      ],
      "modifier": null,
      "number": 1,
      "type": "caption",
      "wordCount": 8,
    },
    Object {
      "content": Array [
        Object {
          "content": "Am I paranoid?",
          "type": "text",
        },
      ],
      "modifier": null,
      "number": 2,
      "type": "caption",
      "wordCount": 3,
    },
    Object {
      "type": "blank",
    },
    Object {
      "captionCount": 1,
      "captionWordCount": 6,
      "dialogueCount": 0,
      "dialogueWordCount": 0,
      "number": 3,
      "sfxCount": 0,
      "speakers": Array [],
      "type": "panel",
    },
    Object {
      "content": "Character looks behind him, thinking he is being followed.",
      "type": "paragraph",
    },
    Object {
      "content": Array [
        Object {
          "content": "I like to call it prepared.",
          "type": "text",
        },
      ],
      "modifier": null,
      "number": 3,
      "type": "caption",
      "wordCount": 6,
    },
    Object {
      "type": "blank",
    },
    Object {
      "captionCount": 1,
      "captionWordCount": 4,
      "dialogueCount": 1,
      "dialogueWordCount": 2,
      "number": 4,
      "sfxCount": 0,
      "speakers": Array [
        "CHARACTER",
      ],
      "type": "panel",
    },
    Object {
      "content": "Character running frantically.",
      "type": "paragraph",
    },
    Object {
      "content": Array [
        Object {
          "content": "Help! Help!",
          "type": "text",
        },
      ],
      "modifier": null,
      "number": 4,
      "speaker": "CHARACTER",
      "type": "dialogue",
      "wordCount": 2,
    },
    Object {
      "content": Array [
        Object {
          "content": "Nobody cares, not here.",
          "type": "text",
        },
      ],
      "modifier": null,
      "number": 5,
      "type": "caption",
      "wordCount": 4,
    },
    Object {
      "captionCount": 1,
      "captionWordCount": 3,
      "dialogueCount": 0,
      "dialogueWordCount": 0,
      "number": 5,
      "sfxCount": 1,
      "speakers": Array [],
      "type": "panel",
    },
    Object {
      "content": "Character is shot in the back.",
      "type": "paragraph",
    },
    Object {
      "content": "BLAM!",
      "modifier": null,
      "number": 6,
      "type": "sfx",
    },
    Object {
      "content": Array [
        Object {
          "content": "Bout damn time.",
          "type": "text",
        },
      ],
      "modifier": null,
      "number": 7,
      "type": "caption",
      "wordCount": 3,
    },
  ],
  "spread": Object {
    "captionCount": 5,
    "captionWordCount": 24,
    "dialogueCount": 1,
    "dialogueWordCount": 2,
    "pageCount": 1,
    "panelCount": 5,
    "sfxCount": 1,
    "speakers": Array [
      "CHARACTER",
    ],
    "type": "spread",
  },
}
`;

exports[`parse aggregate parseRawSpreadChunk multiple panels 1`] = `
Object {
  "children": Array [
    Object {
      "captionCount": 0,
      "captionWordCount": 0,
      "dialogueCount": 0,
      "dialogueWordCount": 0,
      "number": 1,
      "sfxCount": 0,
      "speakers": Array [],
      "type": "panel",
    },
    Object {
      "content": "Establihsing shot of Manhattan. There is a helicopter hovering above a skyscraper.",
      "type": "paragraph",
    },
    Object {
      "type": "blank",
    },
    Object {
      "captionCount": 0,
      "captionWordCount": 0,
      "dialogueCount": 0,
      "dialogueWordCount": 0,
      "number": 2,
      "sfxCount": 0,
      "speakers": Array [],
      "type": "panel",
    },
    Object {
      "content": "The character walks down the street. Hands stuffed in pockets.",
      "type": "paragraph",
    },
    Object {
      "type": "blank",
    },
    Object {
      "captionCount": 0,
      "captionWordCount": 0,
      "dialogueCount": 0,
      "dialogueWordCount": 0,
      "number": 3,
      "sfxCount": 0,
      "speakers": Array [],
      "type": "panel",
    },
    Object {
      "content": "Character looks behind him, thinking he is being followed.",
      "type": "paragraph",
    },
    Object {
      "type": "blank",
    },
  ],
  "spread": Object {
    "captionCount": 0,
    "captionWordCount": 0,
    "dialogueCount": 0,
    "dialogueWordCount": 0,
    "pageCount": 1,
    "panelCount": 3,
    "sfxCount": 0,
    "speakers": Array [],
    "type": "spread",
  },
}
`;
